dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeFactoryImpl.java

```diff
@@ -397,31 +398,68 @@ private ContentType dbSaveUpdate(final ContentType saveType) throws DotDataExcep
     	retType = new ImplClassContentTypeTransformer(retType).from();
     }
 
-    // set up default fields
-    if (oldContentType == null) {
-    	List<Field> fields = new ArrayList<Field>(saveType.fields());
-
-        for (Field ff : retType.requiredFields()) {
-          Optional<Field> optional = fields.stream().filter(x -> ff.variable().equalsIgnoreCase(x.variable())).findFirst();
-          if (!optional.isPresent()) {
-            fields.add(ff);
-          }
+    final List<Field> fields = new ArrayList<>(saveType.fields());
+    for (final Field requiredField : retType.requiredFields()) {
+        Optional<Field> foundField = fields
+                .stream()
+                .filter(x -> requiredField.variable().equalsIgnoreCase(x.variable()))
+                .findFirst();
+        if (!foundField.isPresent()) {
+            fields.add(requiredField);
         }
+    }
 
-        FieldAPI fapi = APILocator.getContentTypeFieldAPI();
-        for (Field f : fields) {
-          f = FieldBuilder.builder(f).contentTypeId(retType.id()).build();
-          try {
-            fapi.save(f, APILocator.systemUser());
-          } catch (DotSecurityException e) {
-            throw new DotStateException(e);
-          }
+    final FieldAPI fapi = APILocator.getContentTypeFieldAPI();
```

Comment:
I am wondering if this code can not be move to the ContentTypeAPI after dbSaveUpdate, it woud be desire to avoid API call into a Factory objects.

If there is not possible, would be ok

Reply:
@jdotcms that call to FieldAPI was already there. Probably I formatted the code and that's it looks like I added it.
The main change I've introduced is the `saveFieldVariables()` method.